src/libraries/Microsoft.VisualBasic.Core/src/Microsoft/VisualBasic/FileIO/SpecialDirectories.vb

```diff
@@ -155,8 +166,9 @@ Namespace Microsoft.VisualBasic.FileIO
             Return FileSystem.NormalizePath(Directory)
         End Function
 
+        <RequiresUnreferencedCode("Cannot statically analyze the passed in type.")>
         Private Shared Function GetWindowsFormsDirectory(typeName As String, propertyName As String) As String
```

Comment:
Could we change this to have a DynamicallyAccessedMembers attribute on the `typeName As String`? And then pass the fully-qualified type in above. That way, when the assembly is present, the linker will see the full type name, and know to preserve properties on the type.

Reply:
I remember I tried that, but then wasn't really sure if it would work since I was still getting the warning due to of course System.Windows.Forms not being present so I left it as RUC for now. I'll change it to be just annotated and suppress the warning for the missing assembly instead.